src/Models/ContainerDeviceBase.ts

```diff
@@ -66,8 +66,8 @@ export abstract class ContainerDeviceBase implements Device {
     return this.componentType;
   }
 
-  async checkPrerequisites(): Promise<boolean> {
-    return true;
+  async checkPrerequisites(): Promise<void> {
```

Comment:
Currently `checkPrerequisites(operation: string): Promise<void>;` is the required method of interface `Component`.
Both `ContainerDeviceBase` and `ArduinoDeviceBase` needs to implement this method.
Is it necessary to move `checkPrerequisite()` to be another interface like `ExtensionDependent`?

For the different method signature:

1. ts allows ignoring extra function parameters. More info see [link](https://stackoverflow.com/questions/51727721/typescript-interface-implementing-doesnt-check-method-parameters). `parameter operation: string` does appears because they are not needed in the implementation of the function. 
2. `eslint` does not allow unused vars.